packages/blocks/src/__internal__/utils/gesture.ts

```diff
@@ -158,36 +165,45 @@ export function initMouseEventHandlers(
       e.preventDefault();
     }
 
-    if (!isDragging)
-      onContainerClick(
-        toSelectionEvent(e, getBoundingClientRect, startX, startY)
-      );
-    else
+    if (isDragging) {
       onContainerDragEnd(
         toSelectionEvent(e, getBoundingClientRect, startX, startY, last)
       );
-
-    const horizontalElement = getClosestHorizontalEditor(e.clientY);
-
-    if (!isInsideBlockContainer(e.target) && horizontalElement) {
-      let containerDom = document.querySelector(
-        '.affine-default-page-block-container'
+    } else {
+      onContainerClick(
+        toSelectionEvent(e, getBoundingClientRect, startX, startY)
       );
-      // TODO adapt edgeless block
-      if (!containerDom) {
-        containerDom = document.querySelector(
-          '.affine-edgeless-page-block-container'
-        );
-      }
 
-      if (containerDom) {
-        const react = containerDom.getBoundingClientRect();
-        if (e.clientX < react.left) {
-          const range = setStartRange(horizontalElement);
-          resetNativeSelection(range);
-        } else {
-          const range = setEndRange(horizontalElement);
-          resetNativeSelection(range);
+      if (!isTitleElement(e.target)) {
```

Comment:
Do you have some alternative place to maintain this part? Adding block-type-specific code to the gesture recognizer looks like code smell or leaky abstraction.